### PR TITLE
Add  HentaiArchive

### PR DESCRIPTION
**Pull request by mr-brune**
Support for tag searching it's difficult to implement for this site 

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Comment by mr-brune** [comment]
will make tag support

**Comment by mr-brune** [comment]
will make tag support

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+class HentaiArchive : ParsedHttpSource() {
+    override val name = "HentaiArchive"
+
+    override val baseUrl = "https://www.hentai-archive.com"
+
+    override val lang = "it"
+
+    private val cdnHeaders = super.headersBuilder()
+        .add("Accept", "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8")
+        .add("Referer", baseUrl) // Replace with the actual referer if needed
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
        .add("Referer", "$baseUrl/")
```

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // Define the array of specific queries
+        // val specialQueries = arrayOf("hentai-senza-censura", "hentai-tette-grosse", "hentai-milf", "hentai-studentesse", "hentai-full-color", "hentai-ita-controllo-mentale", "hentai-ita-bikini", "hentai-culi-grossi", "hentai-pelle-scura", "hentai-insegnanti", "hentai-tradimento", "hentai-mostri", "hentai-sesso-anale", "doujinshi", "hentai-sex-toys", "hentai-superdotati", "hentai-monster-girl", "fumetti-porno", "hentai-ita-harem", "hentai-futanari", "hentai-uomini-maturi", "hentai-ita-domestiche", "hentai-esibizioniste", "hentai-famiglia", "hentai-fantasy", "hentai-femdom", "hentai-follia", "hentai-formose", "hentai-pelose", "hentai-lesbiche", "hentai-yaoi", "hentai-x-ray", "hentai-gola-profonda", "hentai-ita-sesso-estremo", "hentai-gyaru", "hentai-ita-infermiere")
+
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
```
not needed?

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        // Default behavior
+        return GET("$baseUrl/page/$page/?s=$query")
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: use `HttpUrl.Builder` to properly encode the query.

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    override fun searchMangaSelector() = "article.result"
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        // Extract and set the URL without domain, handling potential null values
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: redundant?

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        // Extract and set the URL without domain, handling potential null values
+        element.selectFirst("h2.title a")?.absUrl("href")?.let {
+            setUrlWithoutDomain(it)
+        }
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: url is mandatory so add `!!`

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+        // Extract and set the title, providing a default value if not found
+        title = element.selectFirst("h2.title a")?.text() ?: "Unknown Title"
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
        title = element.selectFirst("h2.title a")!!.text()
```

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+                imageUrl = element.attr("src")
+            }
+
+            // Ensure the URL is absolute
+            if (!imageUrl.startsWith("http")) {
+                imageUrl = element.absUrl("data-src")
+                if (imageUrl.isEmpty()) {
+                    imageUrl = element.absUrl("src")
+                }
+            }
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: can make a function `Element.imgAttr()`, and use `when` block

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+            // Remove the part between '-' and '.jpg' if it exists
+            imageUrl = imageUrl.replace(Regex("-(\\d+x\\d+)(?=\\.jpg)"), "")
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: put regex in class constant so it doesn't get recreated in every loop iteration

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+            Page(index, imageUrl = imageUrl)
+        }.sortedBy { it.index }
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
        }
```
not needed?

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
+
+    companion object {
+        const val PREFIX_SEARCH = "id:"
+    }
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
```
not used?

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt (hunk 3 above)
Comment (mr-brune) [reply src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]: cant do it

Comment (AwkwardPeak7) [reply src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```
val url = "$baseUrl/page/$page/".toHttpUrl().newBuilder()
    .addQueryParameter("s", query)
    .build()
```

File: src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/page/$page/".toHttpUrl().newBuilder()
+            .addQueryParameter("s", query)
+            .build()
+        return GET(url.toString())
```
Comment (AwkwardPeak7) [inline src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt]:
```suggestion
        return GET(url, headers)
```

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**